src/main/java/nova/core/block/component/BlockProperty.java

```diff
@@ -85,8 +85,8 @@ public double getResistance() {
 	 */
 	@UnsidedComponent
 	public static class BlockSound extends Component implements BlockProperty {
-		public BiMap<BlockSoundTrigger, Sound> blockSoundSoundMap = HashBiMap.create();
-		public Optional<BiMap<String, Sound>> customDefinedSounds;
+		private final Map<BlockSoundTrigger, Sound> blockSounds = new HashMap<>();
+		private Optional<Map<String, Sound>> customSounds = Optional.empty();
```

Comment:
Do we need to distinguish between `Optional.empty` and an empty map here? If so, just use the latter and simplify the logic massively.